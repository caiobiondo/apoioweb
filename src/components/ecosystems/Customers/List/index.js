import React, { Component } from 'react';
import CustomersList from './organisms/CustomersList/CustomersList';
import { Main, ActionButtonContainer } from './index.styles';
import RemoveCustomerButton from './organisms/RemoveCustomerButton/RemoveCustomerButton';

class CustomersListWrapper extends Component {
  state = {
    empty: false,
    loading: true,
    selectedCustomers: [],
  };

  onLoadFinished = (empty, loading) => {
    this.setState({ empty: empty, loading: loading });
  };

  onSelectCustomer = customer => {
    const { selectedCustomers } = this.state;

    if (Array.isArray(customer)) {
      const costumers = customer.length !== selectedCustomers.length ? customer : [];
      return this.setState({
        selectedCustomers: [...costumers],
      });
    }

    const newSelected = selectedCustomers.filter(({ id }) => id !== customer.id);

    if (newSelected.length !== selectedCustomers.length) {
      return this.setState({ selectedCustomers: newSelected });
    }

    return this.setState({
      selectedCustomers: [...selectedCustomers, customer],
    });
  };

  onRemoveCustomer = () => {
    this.setState({ selectedCustomers: [] });
  };

  render() {
    const { selectedCustomers, loading, empty } = this.state;
    return (
      <Main loading={loading} empty={empty}>
        {!loading && (
          <ActionButtonContainer empty={empty}>
            <RemoveCustomerButton
              selected={selectedCustomers}
              onRemove={this.onRemoveCustomer}
              isCustomerSelected={selectedCustomers.length}
            />
          </ActionButtonContainer>
        )}
        <CustomersList
          onLoadFinished={this.onLoadFinished}
          onSelect={this.onSelectCustomer}
          selectedCustomers={selectedCustomers}
        />
      </Main>
    );
  }
}

export default CustomersListWrapper;
