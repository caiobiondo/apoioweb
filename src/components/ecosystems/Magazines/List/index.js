import React from 'react';

import Magazines from './organisms/Magazines';

import { getCommercialRegionIdFromUser, getSalesManagementIdFromUser } from 'utils/getUserParams';

const MagazinesWrapper = ({ user, match }) => {
  const region = getCommercialRegionIdFromUser(user);
  const gv = getSalesManagementIdFromUser(user);

  return <Magazines type={match.params.type} region={region} gv={gv} />;
};

export default MagazinesWrapper;
