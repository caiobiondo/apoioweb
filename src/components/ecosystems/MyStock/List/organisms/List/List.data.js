import gql from 'graphql-tag';

const ITEMS_PER_PAGE = 10;

export const MyStockProductsQuery = gql`
  query MyStockProductsQuery($offset: Int!, $limit: Int!, $productName: String) {
    stockProducts(offset: $offset, limit: $limit, productName: $productName) {
      id
      userId
      productCode
      productName
      productDescription
      productImage
      stockQuantity
      productPrice
    }
  }
`;

export const updateQuery = (previousResult, { fetchMoreResult }) => {
  if (!fetchMoreResult) {
    return previousResult;
  }
  return Object.assign({}, previousResult, {
    stockProducts: [...previousResult.stockProducts, ...fetchMoreResult.stockProducts],
  });
};

export const MyStockProductsQueryOptions = {
  options(props) {
    return {
      variables: {
        limit: ITEMS_PER_PAGE,
        offset: 0,
        productName: props.productName,
      },
      forceFetch: true,
    };
  },
  props({ data }) {
    return {
      data,
      loading: data.loading,
      stockProducts: data.stockProducts,
      fetchMore(productName) {
        if (productName) {
          return data.fetchMore({
            variables: {
              offset: 0,
              productName,
            },
            updateQuery: updateQuery,
          });
        }

        return data.fetchMore({
          variables: {
            offset: data.stockProducts.length,
          },
          updateQuery: updateQuery,
        });
      },
    };
  },
};
