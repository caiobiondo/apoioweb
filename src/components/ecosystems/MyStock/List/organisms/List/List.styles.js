import styled from 'styled-components';
import { Responsive } from '@entria/components';
import { gray300, white } from 'styles/colors';
import TableStockCell from '../../molecules/TableStockCell';

export const WrapperStyle = {
  flex: '1 1 0',
  flexDirection: 'column',
  justifyContent: 'center',
  overflowY: 'scroll',
};

export const StockInputWrapper = styled.div``;
// align-itens: flex

export const TableInfoCellWrapper = styled.div`
  ${StockInputWrapper} {
    display: none;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    ${StockInputWrapper} {
      display: flex;
      flex-direction: column;
      align-items: center;
    }
  }
`;

export const TableWrapper = styled.div`
  padding: 20px 20px;

  tr {
    transition: all 0.2s ease;
    background: ${white};
  }

  thead tr th {
    color: ${gray300} !important;
    font-size: 15px !important;
    font-weight: 400 !important;
    text-transform: uppercase;
  }

  tr th:last-child,
  tr td:last-child {
    width: 110px;
  }

  tbody tr:hover {
    background: #f6f6f6;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    tr td,
    tr th {
      display: none;
    }

    tr th:first-child,
    tr td:first-child {
      display: table-cell;
      border-bottom: 1px solid ${gray300};
    }
  }
`;
