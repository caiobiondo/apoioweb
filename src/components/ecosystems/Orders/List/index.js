import React, { Component } from 'react';
import OrdersList from 'components/organisms/OrdersList/OrdersList';
import OrderAddButton from 'components/atoms/OrderAddButton/OrderAddButton';
import { Main, OrderAddButtonContainer } from './index.styles';

class Orders extends Component {
  state = {
    empty: false,
    loading: true,
  };

  onLoadFinished = (empty, loading) => {
    this.setState({ empty: empty, loading: loading });
  };

  render() {
    const { loading, empty } = this.state;
    return (
      <Main loading={loading} empty={empty}>
        {!loading && (
          <OrderAddButtonContainer empty={empty}>
            <OrderAddButton />
          </OrderAddButtonContainer>
        )}
        <OrdersList onLoadFinished={this.onLoadFinished} />
      </Main>
    );
  }
}

export default Orders;
