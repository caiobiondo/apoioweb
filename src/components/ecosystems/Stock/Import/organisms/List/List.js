import React, { Component } from 'react';
import OrdersList from 'components/ecosystems/Orders/List/organisms/OrdersList/OrdersList';
import { Main } from 'components/ecosystems/Orders/List/index.styles';

class StockImportOrdersList extends Component {
  state = {
    empty: false,
    loading: true,
  };

  onLoadFinished = (empty, loading) => {
    this.setState({ empty: empty, loading: loading });
  };

  render() {
    const { loading, empty } = this.state;
    return (
      <Main loading={loading} empty={empty}>
        <OrdersList importing={true} onLoadFinished={this.onLoadFinished} />
      </Main>
    );
  }
}

export default StockImportOrdersList;
