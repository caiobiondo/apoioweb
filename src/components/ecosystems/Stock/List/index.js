import React, { Component } from 'react';
import StockSearch from './organisms/ListSearch';
import StockList from './organisms/ListTable';
import StockAddButton from './organisms/StockAddButton';
import { Main, StockSearchContainer, StockAddButtonContainer } from './index.styles';
import StockAddProductModal from './organisms/StockAddProductModal';
import { gtmPushDataLayerEvent, events } from 'utils/googleTagManager';

class StockListWrapper extends Component {
  state = {
    empty: false,
    loading: true,
    productSearch: '',
    addStockModalOpen: false,
  };

  onLoadFinished = (empty, loading) => {
    this.setState({ empty: empty, loading: loading });
  };

  onSearch = filters => {
    this.setState({ productSearch: filters.name });
  };

  openAddStockModal = () => {
    this.setState({ addStockModalOpen: true });
  };

  handleClose = () => {
    this.setState({ addStockModalOpen: false });
  };

  openAddButton = () => {
    gtmPushDataLayerEvent({
      event: events.EXPAND_ADD_BUTTON,
    });
  };

  closeAddButton = () => {
    gtmPushDataLayerEvent({
      event: events.COLLAPSE_ADD_BUTTON,
    });
  };

  render() {
    const { loading, empty } = this.state;
    return (
      <Main>
        {!loading && (
          <StockSearchContainer>
            <StockSearch onSearch={this.onSearch} />
          </StockSearchContainer>
        )}
        {!loading && (
          <StockAddButtonContainer empty={empty}>
            <StockAddButton
              openAddStockModal={this.openAddStockModal}
              open={this.openAddButton}
              close={this.closeAddButton}
            />
          </StockAddButtonContainer>
        )}
        <StockAddProductModal
          opened={this.state.addStockModalOpen}
          handleClose={this.handleClose}
          user={this.props.user}
        />
        <StockList onLoadFinished={this.onLoadFinished} productSearch={this.state.productSearch} />
      </Main>
    );
  }
}

export default StockListWrapper;
