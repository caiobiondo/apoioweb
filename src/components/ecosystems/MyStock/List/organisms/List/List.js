import React from 'react';
import EmptyList from 'components/molecules/EmptyList/EmptyList';
import { Loading, Paper, Table } from 'natura-ui';
import { MyStockProductsQuery, MyStockProductsQueryOptions } from './List.data';
import withAuthErrorHandler from 'hocs/withAuthErrorHandler/withAuthErrorHandler';
import { graphql } from 'react-apollo';
import { FormattedMessage } from 'react-intl';

import TableStockCell from '../../molecules/TableStockCell';
import TableInfoCell from '../../molecules/TableInfoCell';

import { WrapperStyle, TableWrapper, TableInfoCellWrapper, StockInputWrapper } from './List.styles';

const renderInfoCell = ({ value, row }) => (
  <TableInfoCellWrapper>
    <TableInfoCell product={row} />

    <StockInputWrapper>
      <TableStockCell product={row} />
    </StockInputWrapper>
  </TableInfoCellWrapper>
);

const renderStockCell = ({ value, row }) => <TableStockCell product={row} />;

const TABLE_SCHEMA = {
  columns: ['info', 'stock'],
  style: {
    info: {},
    stock: {},
  },
  renderer: {
    info: renderInfoCell,
    stock: renderStockCell,
  },
  header: {
    info: <FormattedMessage id="stockProductHeaderName" />,
    stock: <FormattedMessage id="stockProductHeaderQuantity" />,
  },
};

const StockList = props => {
  if (props.loading) {
    return <Loading background="transparent" />;
  }

  if (!props.stockProducts) {
    return (
      <EmptyList
        icon="ico_forklift"
        titleId="stockEmptyList"
        descriptionId="stockEmptyListDescription"
      />
    );
  }

  return (
    <Paper style={WrapperStyle}>
      <TableWrapper>
        <Table data={{ ...TABLE_SCHEMA, body: props.stockProducts }} />
      </TableWrapper>
    </Paper>
  );
};

export const StockListWithAuthErrorHandler = withAuthErrorHandler(StockList);

export default graphql(MyStockProductsQuery, MyStockProductsQueryOptions)(
  StockListWithAuthErrorHandler,
);
