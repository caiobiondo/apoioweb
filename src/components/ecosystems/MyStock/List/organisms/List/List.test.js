import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('Stock Product List', () => {
  it('should render the stock product list page', () => {
    expect(shallow(<List />)).toMatchSnapshot();
  });
});
