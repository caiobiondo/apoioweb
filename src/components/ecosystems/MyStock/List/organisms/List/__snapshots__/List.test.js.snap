// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Product List should render the stock product list page 1`] = `
<Liststyles__Wrapper>
  <ListSearch
    onSearch={[Function]}
  />
  <Apollo(WithAuthErrorHandlerComponent)
    productSearch=""
  />
</Liststyles__Wrapper>
`;
